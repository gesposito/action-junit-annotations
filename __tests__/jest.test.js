const { failingCasesFrom, annotationsFrom } = require("../index");

describe("Flat Jest report", () => {
  test("Converts Jest JUnit XML to GitHub annotations", async () => {
    const failingCases = await failingCasesFrom(
      "./test-results/jest-junit-failure.xml"
    );
    expect(failingCases).toMatchSnapshot();

    const annotations = annotationsFrom(failingCases);
    expect(annotations).toMatchSnapshot();
  });
});

describe("Nested Jest report", () => {
  let failingCases;
  test("Converts Jest JUnit XML to failing cases in JSON", async () => {
    failingCases = await failingCasesFrom(
      "./test-results/jest-junit-nested-failure.xml"
    );
    expect(failingCases).toMatchSnapshot();
  });

  test("Converts Jest failing cases in JSON to GitHub annotations", async () => {
    const annotations = annotationsFrom(failingCases);
    expect(annotations).toMatchSnapshot();
  });
});
