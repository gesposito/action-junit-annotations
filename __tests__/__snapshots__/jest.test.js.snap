// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flat Jest report Converts JUnit XML to GitHub annotations 1`] = `
Array [
  Object {
    "classname": "Converts JUnit report to GitHub annotations",
    "failure": "Error: test
at failingCasesFrom (/Users/gesposito/Work/action-junit-annotations/index.js:105:9)
at Object.&lt;anonymous&gt; (/Users/gesposito/Work/action-junit-annotations/index.test.js:4:24)",
    "name": "Converts JUnit report to GitHub annotations",
    "path": "index.test.js",
    "time": "0.015",
  },
]
`;

exports[`Flat Jest report Converts JUnit XML to GitHub annotations 2`] = `
Array [
  Object {
    "annotation_level": "failure",
    "end_line": 0,
    "message": "Error: test
at failingCasesFrom (/Users/gesposito/Work/action-junit-annotations/index.js:105:9)
at Object.&lt;anonymous&gt; (/Users/gesposito/Work/action-junit-annotations/index.test.js:4:24)",
    "path": "index.test.js",
    "start_line": 0,
  },
]
`;

exports[`Nested Jest report Converts JUnit XML to failing cases in JSON 1`] = `
Array [
  Object {
    "classname": "Nested report Converts failing cases in JSON to GitHub annotations",
    "failure": "Error: error
    at Object.&lt;anonymous&gt; (/Users/gesposito/Work/action-junit-annotations/index.test.js:53:11)
    at Object.asyncJestTest (/Users/gesposito/Work/action-junit-annotations/node_modules/jest-jasmine2/build/jasmineAsyncInstall.js:106:37)
    at /Users/gesposito/Work/action-junit-annotations/node_modules/jest-jasmine2/build/queueRunner.js:45:12
    at new Promise (&lt;anonymous&gt;)
    at mapper (/Users/gesposito/Work/action-junit-annotations/node_modules/jest-jasmine2/build/queueRunner.js:28:19)
    at /Users/gesposito/Work/action-junit-annotations/node_modules/jest-jasmine2/build/queueRunner.js:75:41",
    "name": "Nested report Converts failing cases in JSON to GitHub annotations",
    "path": "index.test.js",
    "time": "0.001",
  },
]
`;

exports[`Nested Jest report Converts failing cases in JSON to GitHub annotations 1`] = `
Array [
  Object {
    "annotation_level": "failure",
    "end_line": 0,
    "message": "Error: error
    at Object.&lt;anonymous&gt; (/Users/gesposito/Work/action-junit-annotations/index.test.js:53:11)
    at Object.asyncJestTest (/Users/gesposito/Work/action-junit-annotations/node_modules/jest-jasmine2/build/jasmineAsyncInstall.js:106:37)
    at /Users/gesposito/Work/action-junit-annotations/node_modules/jest-jasmine2/build/queueRunner.js:45:12
    at new Promise (&lt;anonymous&gt;)
    at mapper (/Users/gesposito/Work/action-junit-annotations/node_modules/jest-jasmine2/build/queueRunner.js:28:19)
    at /Users/gesposito/Work/action-junit-annotations/node_modules/jest-jasmine2/build/queueRunner.js:75:41",
    "path": "index.test.js",
    "start_line": 0,
  },
]
`;
